js/data/bucket.js

```diff
@@ -336,36 +335,97 @@ function capitalize(string) {
     return string.charAt(0).toUpperCase() + string.slice(1);
 }
 
-function createAttributes(bucket) {
+function createPaintAttributes(bucket) {
     var attributes = {};
     for (var interfaceName in bucket.programInterfaces) {
-        var interfaceAttributes = attributes[interfaceName] = { layoutAttributes: [], paintAttributes: {} };
-        var layerPaintAttributes = interfaceAttributes.paintAttributes;
+        var layerPaintAttributes = attributes[interfaceName] = {};
 
         for (var c = 0; c < bucket.childLayers.length; c++) {
             var childLayer = bucket.childLayers[c];
-            layerPaintAttributes[childLayer.id] = { enabled: [], disabled: [] };
+            layerPaintAttributes[childLayer.id] = { enabled: [], uniforms: [] };
```

Comment:
I'll switch this to `attributes` / `uniforms`. Zoom + data functions create uniforms